server/services/user.js

```diff
@@ -0,0 +1,77 @@
+import Joi from 'joi';
+import User from '../models/user';
+
+/**
+ * Create user
+ * @param {Object} userObject - The user details
+ * @returns {Promise<User, ValidationError>}
+ */
+function createUser(userObject) {
+  const userSchema = Joi.object().keys({
```

Comment:
It's a little weird that this is essentially duplicate from the route validation. I know the other has to be inside 'body' though. Maybe at some point we can centralize these validators...